### PR TITLE
external authentication api

### PR DESCRIPTION
**Pull request by cryogenian**
[ch11516]

Do we need to test that datasource modules propagates `getAuth` param w/o changing? 🙂 

snapshot `184.0.2-3b3f8f2`

### CODE REVIEW

File: foundation/src/main/scala/quasar/contrib/fs2/pipe.scala [hunk 1]
```diff
@@ -47,4 +47,24 @@ object pipe {
         Stream.chunk(Chunk.ByteBuffer(encoder.encode(cb))))
     }
   }
+
+  def limitSize[F[_]](
```
Comment (alissapajer) [inline foundation/src/main/scala/quasar/contrib/fs2/pipe.scala:51]: was going to point to https://github.com/precog/quasar-datasource-url/blob/master/core/src/main/scala/quasar/datasource/url/GatheredInfo.scala#L86 as a second implementation of this, that could be replaced by this one

looks like it was copied from there though, given the line-by-line diff :)

also, fwiw, multistep will make it so we don't need to impose this limitation anywhere in the url connector anymore. but this is a useful utility, so 👍 to having it here in general.

Comment (cryogenian) [reply foundation/src/main/scala/quasar/contrib/fs2/pipe.scala:51]: Yeah, I copied lots of stuff from ds-url in sdbe pr too :)

**Comment by cryogenian** [comment]
Put 🛑 Until this is ready https://github.com/precog/sdbe/pull/2144

File: api/src/main/scala/quasar/api/auth/ExternalCredentials.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.api.auth
+
+import scala._
+
+trait ExternalCredentials[+F[_]] extends Product with Serializable
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/auth/ExternalCredentials.scala:21]: Why the covariance on the type constructor?

**Pull request merged**